src/evo/mnauth.cpp

```diff
@@ -21,10 +21,10 @@
 
 void CMNAuth::PushMNAUTH(CNode& peer, CConnman& connman, const CBlockIndex* tip)
 {
+    if (!fMasternodeMode) return;
+
     LOCK(activeMasternodeInfoCs);
-    if (!fMasternodeMode || activeMasternodeInfo.proTxHash.IsNull()) {
-        return;
-    }
+    if (activeMasternodeInfo.proTxHash.IsNull()) return;
```

Comment:
It would be a nice refactor to make activeMasternodeInfo into an optional; or some other type that cannot be accessed accidentally.

Reply:
`activeMasternodeInfo` (now `m_info`) is an unconditionally initialized part of `activeMasternodeManager`, which is conditionally initialized. If we had to pass `activeMasternodeManager` as an optional reference, the argument list would be accepting an `std::optional<std::reference_wrapper<CActiveMasternodeManager>>` and unwrapping it would involve `mn_activeman.value().get()` ([`std::optional`](https://en.cppreference.com/w/cpp/utility/optional/value), [`std::reference_wrapper`](https://en.cppreference.com/w/cpp/utility/functional/reference_wrapper/get)).

It can be made less contrived by using `mn_activeman->get()` but regardless, it would require that at the start of every function that uses it more than once, we'd have to setup something like...

```c++
    auto mn_activeman = m_mn_activeman_ref.value().get();
// [...]
    auto mnList = m_dmnman.GetListAtChainTip();
    auto dmn = WITH_LOCK(mn_activeman.cs, return mnList.GetValidMNByCollateral(mn_activeman.GetOutPoint()));
```

Arguably we have to do something similar with all the added assertions already but since the codebase is already used to that approach, we've continued on with it. Maybe we could revisit this approach but IMO it's out of the scope of this PR.